### PR TITLE
fix correction for no_leap calendar in tests

### LINKED ISSUE #4722: Exact Restart tests fail for 765 days
In what will be ctsm5.3.016 I have two 765 day ER tests that fail due to the DRV_RESTART_POINTER file setting (off by a day) being wrong. This uses:

cime6.1.49
cmeps6.0.32

The full description of the case is here for CTSM.

https://github.com/ESCOMP/CTSM/issues/2914

If the logic is wrong, there are likely other test lengths that will fail. But, there is a wide spectrum of test lengths that come with CTSM testing, so many/most must be fine.

### PR DESCRIPTION
**Pull request by jedwards4b**
Test suite:ERP_P64x2_Ld765.f10_f10_mg37.I2000Clm60BgcCrop.derecho_intel.clm-monthly and ERP_P64x2_Ld3285.f10_f10_mg37.I2000Clm60BgcCrop.derecho_intel.clm-monthly
Test baseline:
Test namelist changes:
Test status: bit for bit
Fixes #4722 

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

File: CIME/SystemTests/system_tests_common.py [hunk 1]
```diff
@@ -225,14 +225,15 @@ def _set_restart_interval(
             smon = startdatetime.month
             ryr = restdatetime.year
             rmon = restdatetime.month
-            while ryr > syr:
-                if rmon > 2 and calendar.isleap(ryr):
+            while ryr > syr and (ryr < restdatetime.year or rmon > 2):
```
[the remark this answers is not in the document]
Comment (ekluzek) [reply CIME/SystemTests/system_tests_common.py:228]: Since this has to loop over every year from startdate to restdate I also wonder about how this would do for an especially long period of many years. Practically most likely we aren't going to do restart tests for especially long periods, but for T compsets for example multi-decade restart tests could make sense.

**Review by ekluzek — CHANGES_REQUESTED** [review]
I ran aux_clm for CTSM in ctsm5.3.016 and it's all good. And the tests that failed compare exactly to ctsm5.3.015 as I would expect.

So this fixes this from my perspective. I can create a simple PR to bring this update in.

I do suggest adding some comments around what this section is doing so it's more obvious to future developers. And this really should have some unit testing added for it to test edge cases and longer scenarios. So I'm putting this as requesting changes for those two things, comments and some unittests.

File: CIME/SystemTests/system_tests_common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (billsacks) [reply CIME/SystemTests/system_tests_common.py:228]: I agree that unit tests would be ideal here - covering a few specific cases - if they aren't too hard to put in place. If they are hard to put in place for some reason, then having some comments explaining this logic feels like an acceptable alternative (actually, whether or not there are unit tests).

**Comment by jedwards4b** [comment]
It turned out not to be too hard to add unit tests.  Please feel free to try more cases.

**Comment by billsacks** [comment]
Yes, thank you very much @jedwards4b !
